src/index.ts

```diff
@@ -1,92 +1,27 @@
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
+ * @example
+ * profiler.start();
  */
-import * as assert from 'assert';
-import * as fs from 'fs';
-import * as stream from 'stream';
-import * as util from 'util';
-import * as zlib from 'zlib';
-
-import {serialize} from './builder';
-import {perftools} from './profile';
-
-type HrTimeTuple = [number, number];
-
-const profiler = require('bindings')('cpu_profiler');
-
-function timeToNanos(tuple: HrTimeTuple) {
-  return tuple[0] * 1e9 + tuple[1];
+function start(config: ProfileAgentConfig) {
+  throw new Error('start() is unimplemented');
 }
 
-const durationMillis = 10 * 1000;
-const intervalMillis = 60 * 1000;
-let isActive = false;
-
-function profileInterval() {
-  assert(durationMillis <= intervalMillis);
-  const startDelay = (intervalMillis - durationMillis) * Math.random();
-  setTimeout(function() {
-    const startTime = Date.now();
-    const runName = 'cloud-profile-' + startTime;
-    profiler.startProfiling(runName, true);
-    isActive = true;
-    setTimeout(function() {
-      isActive = false;
-      const result = profiler.stopProfiling(runName);
-      const serialized = serialize(result, startTime * 1e6);
-      const writer = perftools.profiles.Profile.encode(serialized);
-      const buffer = writer.finish();
+module.exports = {
+  start: start
+};
 
-      const outp = fs.createWriteStream(runName + '.pb.gz');
-      const inp = new stream.PassThrough();
-      inp.end(buffer);
-      inp.pipe(zlib.createGzip()).pipe(outp).on('close', function() {
-        setTimeout(
-            profileInterval, intervalMillis - startDelay - durationMillis)
-            .unref();
-      });
-    }, durationMillis).unref();
-  }, startDelay).unref();
+// If the module was --require'd from the command line, start the agent.
+if (module.parent && module.parent.id === 'internal/preload') {
+  module.exports.start();
 }
 
-process.on('exit', function() {
-  if (isActive) {
-    profiler.stopProfiling();
-  }
-});
-process.on('uncaughtException', function() {
-  if (isActive) {
-    profiler.stopProfiling();
-  }
-});
-
-profileInterval();
-
-/*function printNode(node, indent) {
-  indent = indent || 0;
-  var s = sprintf('%5u %s %d #%d %d %s %d', node.hitCount,
-                  Array(indent).join(' '), indent / 2, node.id, node.callUID,
-                  node.functionName, node.scriptId);
-  if (node.url !== '') {
-    s += ' ' + node.url + ':' + node.lineNumber;
-  }
-  if (node.lineTicks)
-    s += ' lineTicks ' + util.inspect(node.lineTicks);
-
-  // TODO: deopt infos
-  // TODO: bailout reason
-  console.log(s);
-  node.children.forEach(function(child) { printNode(child, indent + 2); });
-}*/
+export default {};
```

Comment:
The `module.exports` above conflicts with this ES6 style export. The two systems aren't really compatible and I would suggest using ES6 style. Omit this once you are exporting `start`.